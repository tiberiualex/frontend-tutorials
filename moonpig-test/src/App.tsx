import styled from "styled-components";
import { useAppDispatch, useAppSelector } from "./state/hooks";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  useNavigate,
  useLocation,
} from "react-router-dom";
import Header from "./components/Header/Header";

const AppContainer = styled.div`
  height: 100vh;
  width: 100vw;
`;

const App = () => {
  return (
    <AppContainer className="App">
      <Header />
    </AppContainer>
  );
};

export default App;
