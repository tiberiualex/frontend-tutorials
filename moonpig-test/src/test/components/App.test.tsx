import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../../App";
import { renderWithRouterAndContext } from "../test-utils";

describe("App", () => {
  test("renders app", () => {
    const { container } = renderWithRouterAndContext(<App />);
    expect(container.innerHTML).toMatch("App");
  });

  test("matches snapshot", () => {
    const { container } = renderWithRouterAndContext(<App />);
    expect(container).toMatchSnapshot();
  });
});
