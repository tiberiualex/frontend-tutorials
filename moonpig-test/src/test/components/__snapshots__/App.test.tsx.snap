// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL fwpLXR App"
  >
    <h1>
      Moonpig
    </h1>
  </div>
</div>
`;
